### PR TITLE
Feature/draw results.png in the middle

### PR DESCRIPTION
**Pull request by Farenweh**
Make trainer draw results.png at the end of each epoch, instead of only the end of training. Which may imporve training trace in the case of w/o third-party monitoring.

I have read the CLA Document and I sign the CLA

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR makes minor improvements to documentation and ensures training metrics are plotted at the end of every epoch when enabled.

### 📊 Key Changes
- Added a step to plot training metrics at the end of each epoch if the `plots` option is enabled 📈.
- Made small formatting and clarity improvements across multiple documentation files and example READMEs 📚.

### 🎯 Purpose & Impact
- **Better Training Insights:** Users now get up-to-date metric plots after every training epoch, making it easier to monitor progress and spot issues early.
- **Improved Documentation:** Enhanced docs and examples make it easier for users to follow instructions and understand key concepts.
- **User Experience:** These changes help both new and experienced users get more value and clarity from Ultralytics tools and resources.

✨ Overall, these updates make training monitoring more convenient and the documentation more user-friendly!

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Farenweh, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for helping improve the training experience for the community! 🌟📈

This is an automated response; an Ultralytics engineer will review your PR and assist you soon.

**Comment by Farenweh** [comment]
@Y-T-G Could you please take a look when you have a moment? It adds a new feature to plot metrics at the end of each training epoch, controlled by the plots flag.

**Comment by Farenweh** [comment]
recheck

**Comment by Y-T-G** [comment]
I believe this functionality could be achieved through callbacks if needed. We generally do not want plots every epoch by default due to potential overhead.

**Comment by Farenweh** [comment]
> Collaborator

I see, that's actually a better solution, thanks.

**Comment by glenn-jocher** [comment]
Great to see you're open to the callback approach! You're right that [callbacks](https://docs.ultralytics.com/usage/callbacks/) provide more flexibility for custom plotting behavior without adding overhead to the default training workflow. If you'd like to implement epoch-level plotting, you can create a custom callback that triggers on `on_train_epoch_end` to generate your plots when needed. This keeps the core trainer lean while still enabling the functionality for users who want it.

**Comment by cavalier501** [comment]
> Great to see you're open to the callback approach! You're right that [callbacks](https://docs.ultralytics.com/usage/callbacks/) provide more flexibility for custom plotting behavior without adding overhead to the default training workflow. If you'd like to implement epoch-level plotting, you can create a custom callback that triggers on `on_train_epoch_end` to generate your plots when needed. This keeps the core trainer lean while still enabling the functionality for users who want it.

Hello, I would like to resume training, but only modify the number of epochs to extend or shorten the training process while preserving the other training behavier. Is it possible to make the following modifications to achieve my goal? Thank you for your help.


    target_epochs=500 # None or target_epochs
    ckpt_path="path_to_last.pt"
    if target_epochs is None:
        pass
    else:
        ckpt_=torch.load(ckpt_path)
        ckpt_["train_args"]["epochs"]=target_epochs
        torch.save(ckpt_,ckpt_path)
    model=YOLO(
        ckpt_path
    )
    results = model.train(
        resume=True,
    )

**Comment by glenn-jocher** [comment]
Thanks for the question—no need to edit the checkpoint. To change the total training length while preserving optimizer/LR state, resume and set the target epochs programmatically; a tiny callback is the safest way:

```python
from ultralytics import YOLO

TARGET_EPOCHS = 500  # your new total

def set_total_epochs(trainer):
    trainer.epochs = TARGET_EPOCHS  # adjust planned total before training starts

model = YOLO("path/to/last.pt")
model.add_callback("on_pretrain_routine_start", set_total_epochs)
model.train(resume=True)
```

If the prior run already fully completed and you don’t need the exact optimizer state, start a fresh run from the weights instead (drop `resume=True`). See the Resuming section in the training guide for context: [Resuming interrupted trainings](https://docs.ultralytics.com/modes/train/).
